cumulusci/tasks/apex/batch.py

```diff
@@ -32,83 +34,116 @@ def _run_task(self):
 
         self.logger.info("Job is complete.")
 
-        vals = {
-            key: value
-            for key, value in self.batch.items()
-            if key
-            in {
-                "Id",
-                "Status",
-                "ExtendedStatus",
-                "NumberOfErrors",
-                "JobItemsProcessed",
-                "TotalJobItems",
-            }
-        }
-        if not self.success:
+        summary = self.summarize_batches(self.batches)
+        failed_batches = self.failed_batches(self.batches)
+
+        if failed_batches:
             self.logger.info("There have been some batch failures.")
             self.logger.info("Error values:")
-            self.logger.info(repr(vals))
-            raise SalesforceException("There were import errors: %s" % repr(vals))
-        elif not self.done_for_sure:
+            self.logger.info(repr(failed_batches))
+            raise SalesforceException(
+                f"There were batch errors: {repr(failed_batches)}"
+            )
```

Comment:
It seems redundant to show the failed_batches both in the log and in the exception message. (I see that's how it was before, but this seems like the best time to fix it.)

Reply:
I feel like sometimes logs go one place and exceptions another. Especially in Robot. So I'd rather leave it as-is. Better too much information about errors than too little. I'll keep an eye out in practice as I use it to see if it does consistently duplicate in the relevant contexts.

Reply:
The intention is that exceptions should always be included in logs. I think we do need to do some work to make sure that Robot Framework doesn't hide output from the MetaCI build log.

Reply:
But okay, we can revisit this later.